modules/ppcp-settings/src/Service/SettingsDataManager.php

```diff
@@ -206,14 +203,40 @@ protected function apply_configuration( ConfigurationFlagsDTO $flags ) : void {
 		$this->apply_pay_later_messaging( $flags );
 	}
 
+	/**
+	 * Initialize hooks for gateway synchronization.
+	 *
+	 * @return void
+	 */
+	public function init_hooks() : void {
+		add_action( 'woocommerce_paypal_payments_sync_gateways', array( $this, 'sync_gateway_settings' ) );
+	}
+
+	/**
+	 * Synchronize gateway settings with merchant onboarding choices.
+	 *
+	 * @return void
+	 */
+	public function sync_gateway_settings() : void {
+		$flags = new ConfigurationFlagsDTO();
+
+		$profile_data = $this->onboarding_profile->to_array();
+
+		$flags->is_business_seller = ! ( $profile_data['is_casual_seller'] ?? false );
+		$flags->use_card_payments  = $profile_data['accept_card_payments'] ?? false;
+		$flags->use_subscriptions  = in_array( 'SUBSCRIPTIONS', $profile_data['products'] ?? array(), true );
+
+		$this->toggle_payment_gateways( $flags );
+	}
+
 	/**
 	 * Enables or disables payment gateways depending on the provided
 	 * configuration flags.
 	 *
 	 * @param ConfigurationFlagsDTO $flags Shop configuration flags.
 	 * @return void
 	 */
-	protected function toggle_payment_gateways( ConfigurationFlagsDTO $flags ) : void {
+	public function toggle_payment_gateways( ConfigurationFlagsDTO $flags ) : void {
```

Comment:
Possibly, a left over from some earlier tests 😉

I would recommend making the method protected/private again as it's an internal API